spec/feedx/consumer_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
+
+RSpec.describe Feedx::Consumer do
+  let(:bucket) { BFS::Bucket::InMem.new }
+  let(:klass)  { Feedx::TestCase::Model }
+  before { allow(BFS).to receive(:resolve).and_return(bucket) }
+
+  it 'should reject invalid inputs' do
+    expect do
+      described_class.fetch('mock:///dir/file.txt', klass) {}
+    end.to raise_error(/unable to detect format/)
+  end
+
+  it 'should consume feeds' do
+    url   = mock_produce!
+    count = 0
+    described_class.fetch(url, klass) do |rec|
+      expect(rec).to be_instance_of(klass)
+      count += 1
+    end
+    expect(count).to eq(300)
+  end
+
+  it 'should perform conditionally' do
+    url   = mock_produce! last_modified: Time.at(1515151515)
+
+    count = 0
+    described_class.fetch(url, klass, if_modified_since: Time.at(1414141414)) {|_| count += 1 }
```

Comment:
@ajn @vrih one of the problems I have compared to Go's Consumer version is that Ruby Consumers are not long-living. In Go, we just store the last modified timestamp when pulling the feed the first time and then only consume if there have been any remote changes. In Ruby, we don't keep consumers "around", although we probably could, e.g.:

```
module Clockwork 
  FEED_X_CONSUMER = Feedx::Consumer.new(ENV.fetch('SOME_URL'), SomeModel)

  every 10.minutes do
   FEED_X_CONSUMER.fetch do |rec|
     # ...
   end
  end
```

Reply:
Although, I am not sure how well that would work with jobs. Ideas? Thoughts?

Reply:
Don't we just need an easy way to access the timestamp of the last time the fleiss job ran and pass that through?

Reply:
Not really. When we produce our feed, we are doing something like:

```
Producer.perform ..., last_modified: Model.max(:updated_at)
```

When successful, that `CUSTOM_TIME` is then stored as the revision of the feed in the feed metadata. The timestamp of the last Worker run will certainly not match the revision. The ideal process would be something like:

1. Run `Consumer.fetch`, consume and remember the remote revision.
2. Run `Consumer.fetch ... if_modified: REVISION` which would skip if the passed value still matches the remote one 

The challenge is: how to best remember that revision between runs.

Reply:
We also cannot just use a global Consumer, as it won't be shared across worker processes. I assume there is no way around some sort of persistence layer to accomplish this.